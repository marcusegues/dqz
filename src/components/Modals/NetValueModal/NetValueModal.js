// @flow
import React from 'react';
import type { ComponentType } from 'react';
// $FlowFixMe
import { translate } from 'react-i18next';
import { AppModal } from '../AppModal';
import { CardHeaderSubText } from '../../QuestionAnswer/Cards/subcomponents/CardHeaderSubText';
import { ModalCard } from '../ModalCard';
import { ModalCloseText } from '../ModalCloseText';
import type { TFunction } from '../../../types/generalTypes';
import { BASE_GREY } from '../../../styles/colors';

const ownStyles = {
  pickerCard: {
    width: '100%',
  },

  text: {
    color: BASE_GREY,
    lineHeight: 20,
    marginTop: 16,
  },
};
type NetValueModalInnerInnerProps = {
  modalVisible: boolean,
  toggleModalVisible: () => void,
};

export const NetValueModalInner = ({
  t,
  modalVisible,
  toggleModalVisible,
}: NetValueModalInnerInnerProps & { t: TFunction }) => (
  <AppModal
    modalVisible={modalVisible}
    animationIn="slideInLeft"
    animationOut="slideOutLeft"
  >
    <ModalCard style={ownStyles.pickerCard}>
      <CardHeaderSubText
        text={t('netValueInfoModal:mainText')}
        style={ownStyles.text}
      />
      <CardHeaderSubText
        text={t('netValueInfoModal:calculations')}
        style={ownStyles.text}
      />
    </ModalCard>
    <ModalCloseText
      onModalHide={toggleModalVisible}
      text={t('closeModalText')}
    />
  </AppModal>
);

export const NetValueModal = (translate(['modal', 'netValueInfoModal'])(
  NetValueModalInner
): ComponentType<NetValueModalInnerInnerProps>);
