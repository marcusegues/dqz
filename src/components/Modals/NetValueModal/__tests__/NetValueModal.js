import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { NetValueModal } from '../../NetValueModal/NetValueModal';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

describe('NetValueModal', () => {
  test('NetValueModal renders according to snapshot', () => {
    const component = renderer.create(<NetValueModal />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('shallow rendering with `modalVisible=true`', () => {
    const wrapper = shallow(
      <NetValueModal modalVisible={clickFn} toggleModalVisible={clickFn} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
