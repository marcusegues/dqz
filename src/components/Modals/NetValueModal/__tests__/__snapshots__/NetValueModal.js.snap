// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetValueModal NetValueModal renders according to snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onDismiss={[Function]}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
    testID={undefined}
  />
  <View
    collapsable={undefined}
    onDismiss={[Function]}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFF",
            "borderRadius": 3,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "width": "85%",
          },
          Object {
            "elevation": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2,
          },
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#757575",
            "fontFamily": "roboto_regular",
            "fontSize": 14,
            "lineHeight": 20,
            "marginTop": 16,
            "paddingHorizontal": 31.388235294117646,
          }
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#757575",
            "fontFamily": "roboto_regular",
            "fontSize": 14,
            "lineHeight": 20,
            "marginTop": 16,
            "paddingHorizontal": 31.388235294117646,
          }
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 50.285714285714285,
            "padding": 23.57142857142857,
            "position": "absolute",
          },
          Object {},
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "roboto_regular",
            "fontSize": 22,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</Modal>
`;

exports[`NetValueModal shallow rendering with \`modalVisible=true\` 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NetValueModalInner
    modalVisible={[Function]}
    t={[Function]}
    toggleModalVisible={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "animationIn": "slideInLeft",
      "animationOut": "slideOutLeft",
      "children": Array [
        <ModalCard
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <CardHeaderSubText
            style={
              Object {
                "color": "#757575",
                "lineHeight": 20,
                "marginTop": 16,
              }
            }
            text=""
          />
          <CardHeaderSubText
            style={
              Object {
                "color": "#757575",
                "lineHeight": 20,
                "marginTop": 16,
              }
            }
            text=""
          />
        </ModalCard>,
        <Unknown
          onModalHide={[Function]}
          text=""
        />,
      ],
      "modalVisible": [Function],
      "onDismiss": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeaderSubText
              style={
                Object {
                  "color": "#757575",
                  "lineHeight": 20,
                  "marginTop": 16,
                }
              }
              text=""
            />,
            <CardHeaderSubText
              style={
                Object {
                  "color": "#757575",
                  "lineHeight": 20,
                  "marginTop": 16,
                }
              }
              text=""
            />,
          ],
          "style": Object {
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "color": "#757575",
                "lineHeight": 20,
                "marginTop": 16,
              },
              "text": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "color": "#757575",
                "lineHeight": 20,
                "marginTop": 16,
              },
              "text": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onModalHide": [Function],
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "animationIn": "slideInLeft",
        "animationOut": "slideOutLeft",
        "children": Array [
          <ModalCard
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <CardHeaderSubText
              style={
                Object {
                  "color": "#757575",
                  "lineHeight": 20,
                  "marginTop": 16,
                }
              }
              text=""
            />
            <CardHeaderSubText
              style={
                Object {
                  "color": "#757575",
                  "lineHeight": 20,
                  "marginTop": 16,
                }
              }
              text=""
            />
          </ModalCard>,
          <Unknown
            onModalHide={[Function]}
            text=""
          />,
        ],
        "modalVisible": [Function],
        "onDismiss": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardHeaderSubText
                style={
                  Object {
                    "color": "#757575",
                    "lineHeight": 20,
                    "marginTop": 16,
                  }
                }
                text=""
              />,
              <CardHeaderSubText
                style={
                  Object {
                    "color": "#757575",
                    "lineHeight": 20,
                    "marginTop": 16,
                  }
                }
                text=""
              />,
            ],
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "color": "#757575",
                  "lineHeight": 20,
                  "marginTop": 16,
                },
                "text": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "color": "#757575",
                  "lineHeight": 20,
                  "marginTop": 16,
                },
                "text": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onModalHide": [Function],
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
